pkg/agent/datamodel/types.go

```diff
@@ -616,29 +616,23 @@ type SysctlConfig struct {
 type AgentPoolProfile struct {
-	Name                   string               `json:"name"`
-	VMSize                 string               `json:"vmSize"`
-	OSDiskSizeGB           int                  `json:"osDiskSizeGB,omitempty"`
-	KubeletDiskType        KubeletDiskType      `json:"kubeletDiskType,omitempty"`
-	DNSPrefix              string               `json:"dnsPrefix,omitempty"`
-	OSType                 OSType               `json:"osType,omitempty"`
-	Ports                  []int                `json:"ports,omitempty"`
-	AvailabilityProfile    string               `json:"availabilityProfile"`
-	ScaleSetPriority       string               `json:"scaleSetPriority,omitempty"`
-	ScaleSetEvictionPolicy string               `json:"scaleSetEvictionPolicy,omitempty"`
-	StorageProfile         string               `json:"storageProfile,omitempty"`
-	DiskSizesGB            []int                `json:"diskSizesGB,omitempty"`
-	VnetSubnetID           string               `json:"vnetSubnetID,omitempty"`
-	Subnet                 string               `json:"subnet"`
-	Distro                 Distro               `json:"distro,omitempty"`
-	CustomNodeLabels       map[string]string    `json:"customNodeLabels,omitempty"`
-	PreprovisionExtension  *Extension           `json:"preProvisionExtension"`
-	KubernetesConfig       *KubernetesConfig    `json:"kubernetesConfig,omitempty"`
-	AvailabilityZones      []string             `json:"availabilityZones,omitempty"`
-	VnetCidrs              []string             `json:"vnetCidrs,omitempty"`
-	WindowsNameVersion     string               `json:"windowsNameVersion,omitempty"`
-	CustomKubeletConfig    *CustomKubeletConfig `json:"customKubeletConfig,omitempty"`
-	CustomLinuxOSConfig    *CustomLinuxOSConfig `json:"customLinuxOSConfig,omitempty"`
+	Name                  string               `json:"name"`
+	VMSize                string               `json:"vmSize"`
+	KubeletDiskType       KubeletDiskType      `json:"kubeletDiskType,omitempty"`
+	DNSPrefix             string               `json:"dnsPrefix,omitempty"`
+	OSType                OSType               `json:"osType,omitempty"`
+	Ports                 []int                `json:"ports,omitempty"`
+	AvailabilityProfile   string               `json:"availabilityProfile"`
+	StorageProfile        string               `json:"storageProfile,omitempty"`
+	VnetSubnetID          string               `json:"vnetSubnetID,omitempty"`
+	Distro                Distro               `json:"distro,omitempty"`
+	CustomNodeLabels      map[string]string    `json:"customNodeLabels,omitempty"`
+	PreprovisionExtension *Extension           `json:"preProvisionExtension"`
+	KubernetesConfig      *KubernetesConfig    `json:"kubernetesConfig,omitempty"`
+	VnetCidrs             []string             `json:"vnetCidrs,omitempty"`
+	WindowsNameVersion    string               `json:"windowsNameVersion,omitempty"`
+	CustomKubeletConfig   *CustomKubeletConfig `json:"customKubeletConfig,omitempty"`
```

Comment:
I know we need these two fields, but in RP PR the agentbaker datamodel did not contain these two fields, I may have looked at an older PR though.

Reply:
Thanks for your review, Adil. Are you referring to fields "WindowsNameVersion" and "CustomKubeletConfig"? I found both of them in our RP code.